pages/candy-machines/index.tsx

```diff
@@ -1,67 +1,68 @@
 /* eslint-disable react-hooks/exhaustive-deps */
 import { useWallet } from '@solana/wallet-adapter-react'
 import { CandyMachineList, CreateCandyMachine, Popup, SearchBar, RefreshButton } from 'components'
-import { useRPC } from 'hooks'
+import { useNotification, useRPC } from 'hooks'
 import { candyMachinesState, candyMachineSearchState } from 'lib/recoil-store/atoms'
 import { fetchCandyMachineAccounts } from 'lib/utils'
 import type { NextPage } from 'next'
 import Head from 'next/head'
 import { useEffect, useState } from 'react'
-import { useRecoilState } from 'recoil'
+import { useRecoilState, useRecoilValue } from 'recoil'
 import { Button, Spinner } from '@primer/react'
-import { useRecoilValue, useSetRecoilState } from 'recoil'
+import { NotificationType } from 'lib/interfaces'
 
 const ManageCandyMachines: NextPage = () => {
+    const { connection } = useRPC()
+    const { publicKey } = useWallet()
+    const { addNotification } = useNotification()
+
     const [accounts, setAccounts] = useRecoilState(candyMachinesState)
     const [isLoading, setIsLoading] = useState(false)
     const [isRefreshLoading, setIsRefreshLoading] = useState(false)
-    const [error, setError] = useState(false)
-    const { connection } = useRPC()
-    const { publicKey } = useWallet()
+    const [error, setError] = useState('')
     const [isCreateOpen, setIsCreateOpen] = useState(false)
     const [searchValue, setSearchValue] = useRecoilState(candyMachineSearchState)
     const searchInput = useRecoilValue(candyMachineSearchState)
-    const setCandyMachines = useSetRecoilState(candyMachinesState)
 
     const searchResults = accounts.filter((account) => {
         return account.toLowerCase().includes(searchInput.trim().toLowerCase())
     })
+
     const fetchAccounts = async () => {
         if (!connection) return
+        setIsRefreshLoading(true)
         try {
             const accounts = await fetchCandyMachineAccounts(connection, publicKey!)
             setAccounts(accounts)
-            setError(false)
+            setError('')
         } catch (err) {
             console.error(err)
-        }
-    }
-
-    const refreshCandyMachines = async () => {
-        if (!connection) return
-        setIsRefreshLoading(true)
-        try {
-            const candyMachines = await fetchCandyMachineAccounts(connection, publicKey!)
-            setCandyMachines(candyMachines)
-        } catch (e) {
-            console.error(e)
+            setError((err as Error).message)
         }
         setIsRefreshLoading(false)
     }
 
     useEffect(() => {
         if (!connection) return
-
-        setError(false)
+        setError('')
         setIsLoading(true)
-
         fetchAccounts()
-            .catch(() => setError(true))
+            .catch((e: any) => setError((e as Error).message))
             .finally(() => {
                 setIsLoading(false)
             })
     }, [connection])
 
+    useEffect(() => {
```

Comment:
not sure this error management here makes sense, you are setting `error` but you are really not using it for anything. I would not wrap the notification in this `useEffect`, actually you might not even need the error state, you can simply create notifications inside the catch